invenio_sip2/server.py

```diff
@@ -19,42 +19,48 @@
 
 import re
 import selectors
+import signal
 import socket
 import traceback
 
 from flask import current_app
 
 from .api import Message
 from .errors import InvalidSelfCheckMessageError
-from .models import SelfcheckClient
 from .proxies import current_sip2
+from .records.record import Client, Server
 
 
 class SocketServer:
     """Socket server."""
 
     selector = selectors.DefaultSelector()
-    clients = {}
 
-    def __init__(self, host='0.0.0.0', port=3004, **kwargs):
+    def __init__(self, name, host='0.0.0.0', port=3004, **kwargs):
         """Constructor."""
+        self.server_name = name
         self.host = host
         self.port = port
         self.remote_app = kwargs.pop('remote')
-        lsock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        self.server = Server.create(data=vars(self))
+        sock = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
         # Avoid bind() exception: OSError: [Errno 48] Address already in use
-        lsock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
-        lsock.bind((self.host, self.port))
-        lsock.listen()
+        sock.setsockopt(socket.SOL_SOCKET, socket.SO_REUSEADDR, 1)
+        sock.bind((self.host, self.port))
+        sock.listen()
         # TODO : use another logging method
         print("listening on", (self.host, self.port))
-        lsock.setblocking(False)
+        signal.signal(signal.SIGINT, self.handler_stop_signals)
+        signal.signal(signal.SIGTERM, self.handler_stop_signals)
+        sock.setblocking(False)
         self.selector.register(
-            lsock, selectors.EVENT_READ | selectors.EVENT_WRITE, data=None)
+            sock, selectors.EVENT_READ | selectors.EVENT_WRITE, data=None)
 
     def run(self):
         """Run socket server."""
         try:
+            self.server['status'] = 'running'
```

Comment:
For next PR : maybe a  class to list all possible server state instead of just string

Reply:
todo added